src/main/java/core/basesyntax/data/ShopService.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.data;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.util.FileWriter;
+import core.basesyntax.util.FileWriterImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopService(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    public void process(List<FruitTransaction> transactions) {
+        Map<String, Integer> inventory = new HashMap<>();
+
+        for (FruitTransaction transaction : transactions) {
+            operationStrategy.execute(transaction.getOperation(), transactions, inventory);
```

Comment:
The `execute` method in `OperationStrategy` should be called with the correct parameters. Ensure that the `transactions` list is being used correctly within the `execute` method. It might be more appropriate to pass only the current transaction instead of the entire list.